embulk-core/src/main/java/org/embulk/exec/PreviewExecutor.java

```diff
@@ -174,10 +176,15 @@ public void add(Page page)
         @Override
         public void finish()
         {
+            if (res != null) {
+                // if PreviewResult object already exists, we should reuse it and throw PreviewedNoticeError.
```

Comment:
Just my impression is that explicitly failing may be better if `finish()` twice is not an expected behavior. If this is a normal executor, we may want to fallback. But, this is preview.

It's just my impression, though.